tuf/repository_lib.py

```diff
@@ -1530,18 +1583,27 @@ def generate_timestamp_metadata(snapshot_filename, version, expiration_date,
   # This check ensures arguments have the appropriate number of objects and
   # object types, and that all dict keys are properly named.
   # Raise 'securesystemslib.exceptions.FormatError' if the check fails.
-  securesystemslib.formats.PATH_SCHEMA.check_match(snapshot_filename)
+  securesystemslib.formats.PATH_SCHEMA.check_match(snapshot_file_path)
   tuf.formats.METADATAVERSION_SCHEMA.check_match(version)
   securesystemslib.formats.ISO8601_DATETIME_SCHEMA.check_match(expiration_date)
   securesystemslib.formats.NAME_SCHEMA.check_match(repository_name)
+  securesystemslib.formats.BOOLEAN_SCHEMA.check_match(use_length)
+  securesystemslib.formats.BOOLEAN_SCHEMA.check_match(use_hashes)
 
-  # Retrieve the versioninfo of the Snapshot metadata file.
   snapshot_fileinfo = {}
-  length, hashes = securesystemslib.util.get_file_details(snapshot_filename,
+
+  length, hashes = securesystemslib.util.get_file_details(snapshot_file_path,
       tuf.settings.FILE_HASH_ALGORITHMS, storage_backend)
+
+  length = (use_length and length) or None
+  hashes = (use_hashes and hashes) or None
+
+  _, snapshot_filename = os.path.split(snapshot_file_path)
+  # Retrieve the versioninfo of the Snapshot metadata file.
   snapshot_version = get_metadata_versioninfo('snapshot', repository_name)
-  snapshot_fileinfo[SNAPSHOT_FILENAME] = \
-    tuf.formats.make_fileinfo(length, hashes, version=snapshot_version['version'])
+  snapshot_fileinfo[snapshot_filename] = \
+      tuf.formats.make_metadata_fileinfo(snapshot_version['version'],
+          length, hashes)
```

Comment:
I don't believe this necessary? The function worked before by using the constant `SNAPSHOT_FILENAME` why would do we need to change away from that? I don't think there's any need for the `snapshot_file_path` argument. AFAICS the one place that wasn't passing in 'snapshot.json' was passing 'path/to/snapshot.json' which your change here will strip down to 'snapshot.json'. Am I misunderstanding?

Reply:
`snapshot_file_path` is needed to calculate the hashes and length of the snapshot file by using `get_file_details` from securesyslib: https://github.com/secure-systems-lab/securesystemslib/blob/9b3a78e998412c39418efb29f411591b20cdd236/securesystemslib/util.py#L45
which has the argument `filepath` which is the absolute path to the file in our case to the snapshot file.

Here we can't remove the snapshot_file_path as we did with the target_filename in the `generate_snapshot_metadata` because in `generate_snapshot_metadata` we have the extra `metadata_directory` argument which we can use to create the absolute path needed for `get_file_details`.